src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -561,8 +561,13 @@ public void UpdatePackageSource(PackageSource source, bool updateCredentials, bo
         {
             if (string.Equals(newSource.Name, existingSource.Name, StringComparison.OrdinalIgnoreCase))
             {
-                if ((!string.Equals(newSource.Source, existingSource.Source, StringComparison.OrdinalIgnoreCase) ||
-                    newSource.ProtocolVersion != existingSource.ProtocolVersion) && newSource.IsPersistable)
+                if ((newSource.ProtocolVersion != existingSource.ProtocolVersion) && newSource.IsPersistable)
+                {
+                    Settings.Remove(ConfigurationConstants.PackageSources, existingSource.AsSourceItem());
```

Comment:
@dotnokato 
Why we're doing `remove` here? But not in other case, what is the diff?

Reply:
@erdembayar 
I used `remove` here as a way to work around issue with equality operator including ProtocolVersion in comparison. As a result `AddOrUpdate` was adding new line in config instead of updating the current one if protocol version value was being edited.
When changing the code, I've left the original logic (using only `AddOrUpdate`) for the path that was working correctly. I only used the `Remove`+`AddOrUpdate` in the one scenario that was necessary.

Come to think of it, I think those two cases can be handled together by only having the `Remove` branch.
Shall I update the code?

Reply:
I've combined those two code paths and pushed rebased code.

Reply:
>I used remove here as a way to work around issue with equality operator including ProtocolVersion in comparison.

There are some people who are very particular about their config files, so I'm very confident that we'll get bug reports that the command has changed the order of sources, as we can see in the tests in this PR.

It's not my intention to add scope creep to this PR, but given the config file has an attribute `key`, it's not obvious to me why anything of the fields other than `key` (which I think is `PackageSource.Name`) should matter to `Settings.AddOrUpdate`.  Perhaps that should be changed, rather than this remove-add workaround?

Reply:
I've pushed change that retains the original order of sources in config file. It is using `Remove`+`AddOrUpdate` hack, though.

@zivkan, original code submitted in for this PR contained change dropping `protocolVersion` from equality comparer in `SourceItem.cs`. It was reverted after [Team Review comment](https://github.com/NuGet/NuGet.Client/pull/5002#discussion_r1072625616).

If you believe that this is better course of action and other reviewers agree, I am happy to rework the code.

Reply:
Thanks for linking to the other comment. Unfortunately, I'm in a radically different time zone than the rest of the team, so I'm unable to participate in most of their meetings, like their PR review meeting, so I don't know what they discussed. I have some questions/objections to comments in that thread, so I'll reply there.